modules/popup/react/lib/Popper.tsx

```diff
@@ -65,10 +65,7 @@ export interface PopperProps extends React.HTMLAttributes<HTMLDivElement> {
 }
 
 export const Popper = React.forwardRef<HTMLDivElement, PopperProps>(
-  ({portal = true, open = true, ...elemProps}: PopperProps, forwardRef) => {
-    const localRef = React.useRef<HTMLDivElement>(null);
-    const ref = (forwardRef || localRef) as React.RefObject<HTMLDivElement>;
-
```

Comment:
This was a whoops. `forwardRef` could be a function if it was used in a Class component. In this case, `forwardRef` would be type narrowed to a `React.RefObject` (subtype of `React.Ref`). This was caught as a contravariant violation on line 122 which is totally valid. It required me to fix this possible runtime error by refactoring how `usePopupStack` works with refs and accept `React.Ref` instead of only `React.RefObject`.